core/src/sql/statements/alter/table.rs

```diff
@@ -0,0 +1,211 @@
+};
+use crate::sql::{Idiom, Kind, TableType};
+use derive::Store;
+use reblessive::tree::Stk;
+use revision::revisioned;
+use serde::{Deserialize, Serialize};
+use std::fmt::{self, Display, Write};
+use std::ops::Deref;
+use std::sync::Arc;
+
+#[revisioned(revision = 1)]
+#[derive(Clone, Debug, Default, Eq, PartialEq, PartialOrd, Serialize, Deserialize, Store, Hash)]
+#[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
+#[non_exhaustive]
+pub struct AlterTableStatement {
+	pub name: Ident,
+	pub if_exists: bool,
+	pub drop: Option<bool>,
+	pub full: Option<bool>,
+	pub permissions: Option<Permissions>,
+	pub changefeed: Option<Option<ChangeFeed>>,
+	pub comment: Option<Option<Strand>>,
+	pub kind: Option<TableType>,
+}
+
+impl AlterTableStatement {
+	pub(crate) async fn compute(
+		&self,
+		stk: &mut Stk,
+		ctx: &Context<'_>,
+		opt: &Options,
+		doc: Option<&CursorDoc<'_>>,
+	) -> Result<Value, Error> {
+		// Allowed to run?
+		opt.is_allowed(Action::Edit, ResourceKind::Table, &Base::Db)?;
+		// Fetch the transaction
+		let txn = ctx.tx();
+		// Get the table definition
+		let mut dt = match txn.get_tb(opt.ns()?, opt.db()?, &self.name).await {
+			Ok(tb) => tb.deref().clone(),
+			Err(Error::TbNotFound {
+				..
+			}) if self.if_exists => return Ok(Value::None),
+			Err(v) => return Err(v),
+		};
+		// Process the statement
+		let key = crate::key::database::tb::new(opt.ns()?, opt.db()?, &self.name);
+		if let Some(ref drop) = &self.drop {
+			dt.drop = *drop;
+		}
+		if let Some(ref full) = &self.full {
+			dt.full = *full;
+		}
+		if let Some(ref permissions) = &self.permissions {
+			dt.permissions = permissions.clone();
+		}
+		if let Some(ref changefeed) = &self.changefeed {
+			dt.changefeed = *changefeed;
+		}
+		if let Some(ref comment) = &self.comment {
+			dt.comment = comment.clone();
+		}
+		if let Some(ref kind) = &self.kind {
+			dt.kind = kind.clone();
+		}
+
+		txn.set(key, &dt).await?;
+		// Add table relational fields
+		if let TableType::Relation(rel) = &dt.kind {
+			// Set the `in` field as a DEFINE FIELD definition
+			{
+				let key = crate::key::table::fd::new(opt.ns()?, opt.db()?, &self.name, "in");
+				let val = rel.from.clone().unwrap_or(Kind::Record(vec![]));
+				txn.set(
+					key,
+					DefineFieldStatement {
+						name: Idiom::from(IN.to_vec()),
+						what: self.name.to_owned(),
+						kind: Some(val),
+						..Default::default()
+					},
+				)
+				.await?;
+			}
+			// Set the `out` field as a DEFINE FIELD definition
+			{
+				let key = crate::key::table::fd::new(opt.ns()?, opt.db()?, &self.name, "out");
+				let val = rel.to.clone().unwrap_or(Kind::Record(vec![]));
+				txn.set(
+					key,
+					DefineFieldStatement {
+						name: Idiom::from(OUT.to_vec()),
+						what: self.name.to_owned(),
+						kind: Some(val),
+						..Default::default()
+					},
+				)
+				.await?;
+			}
+		}
+		// Clear the cache
+		txn.clear();
+		// Record definition change
+		if dt.changefeed.is_some() {
```

Comment:
Instead of referencing `dt`, should it reference `self`, so that this only runs when the table's changefeed policy was actually adjusted?